analyses/fusion_filtering/04-project-specific-filtering.Rmd

```diff
@@ -121,8 +125,65 @@ fusion_caller.summary <- fusion_calls %>%
 #to add aggregated caller from fusion_caller.summary
 fusion_calls<-fusion_calls %>% 
   left_join(fusion_caller.summary,by=(c("Sample","FusionName","Fusion_Type"))) %>%
-  dplyr::select(-JunctionReadCount,-SpanningFragCount,-Confidence,-LeftBreakpoint,-RightBreakpoint) %>% unique()
+  unique()
+```
+
+### Idenitify kinase domain retention status
+
+Kinase domainIDs are obtained pfam by a simple grep "kinase" in their Name
+
+```{r}
+# identify kinase domain from bioMartPfam dataframe provided with annoFuse
+bioMartDataPfam <- readRDS(system.file("extdata", "pfamDataBioMart.RDS", package = "annoFuse"))
+
+# look for domains with "kinase" in their domain Name
+kinaseid<-unique(bioMartDataPfam[grep("kinase",bioMartDataPfam$NAME),
+                                 c("pfam_id","NAME")] ) 
+
+kinaseid
+```
+
+Through annoFuse::fusion_driver domain retention status for given kinase pfamID is being added per Gene1A (5` Gene) and Gene1B (3` Gene)
+```{r}
+fusion_calls <- annoFuse::fusion_driver(fusion_calls,
+                                  annotated = TRUE,
+                                  checkDomainStatus=TRUE,
+                                  # check status for given pfamID
+                                  domainsToCheck=kinaseid$pfam_id,
+                                  # we don't want to filter for putative driver fusion yet
+                                  filterPutativeDriver = FALSE
+                                  )
+```
 
+### Adding check to see if reciprocal fusion exists
+
+```{r}
+
+# check for fusions have reciprocal fusions in the same Sample
+# works only for GeneY -- GeneX ; GeneX -- GeneY matches
+recirpocal_fusion <- function(FusionName,Sample,standardFusioncalls ){
+  Gene1A <- strsplit(FusionName,"--")[[1]][1]
```

Comment:
Can you remind me why we're not looking at the `Gene2A` and `Gene2B` here?

Reply:
For intergenic fusions, which has Gene1A/Gene2A--Gene1B or Gene1A--Gene1B/Gene2B and similar fusions it get's a little complicated since then we need to check the distance is the same between Gene1A/Gene2A in the reciprocal so I've just stuck to fusions between genes.